src/components/tree/phyloTree/helpers.js

```diff
@@ -33,19 +34,20 @@ export const applyToChildren = (phyloNode, func) => {
  * of nodes in a rectangular tree.
  * If `yCounter` is undefined then we wish to hide the node and all descendants of it
  * @param {PhyloNode} node
+ * @param {function} getDisplayOrder
  * @param {int|undefined} yCounter
  * @sideeffect modifies node.displayOrder and node.displayOrderRange
  * @returns {int|undefined} current yCounter after assignment to the tree originating from `node`
  */
-export const setDisplayOrderRecursively = (node, yCounter) => {
+export const setDisplayOrderRecursively = (node, getDisplayOrder, yCounter) => {
   const children = node.n.children; // (redux) tree node
   if (children && children.length) {
     for (let i = children.length - 1; i >= 0; i--) {
-      yCounter = setDisplayOrderRecursively(children[i].shell, yCounter);
+      yCounter = setDisplayOrderRecursively(children[i].shell, getDisplayOrder, yCounter);
     }
   } else {
-    node.displayOrder = (node.n.fullTipCount===0 || yCounter===undefined) ? yCounter : ++yCounter;
-    node.displayOrderRange = [yCounter, yCounter];
+    node.displayOrder = (node.n.fullTipCount===0 || yCounter===undefined) ? yCounter : yCounter + getDisplayOrder(node);
```

Comment:
Previously, `yCounter` referred to the previous tip's display order, and we incremented it by one every time we visited a new terminal node. But now things are a bit stranger (IMO):

* `yCounter` is not incremented except for when we jump into a new subtree (via `yCounter+=spaceBetweenSubtrees;`)
* `getDisplayOrder` is a closure and returns the display order for this node, itself keeping track of the previous tip's display order etc.
* so `yCounter` is more like an offset value that changes across subtrees.

I _think_ we can get rid of `getDisplayOrder` entirely and inline it within the existing two functions, using argument passing instead of closures. `yCounter` would then continue to represent the previous tip's value. 

Note that there are a few strange bugs around trees with many subtrees because `yCounter` gets large (lots of `yCounter+=spaceBetweenSubtrees;` calls) and thus the `yPerFocused` and `yPerUnfocused` values are smaller than they should be. I think it'd be clarifying to compute all these values within `setDisplayOrder` so we can know ahead of time the range of display order values we'll generate.

I'm happy to test out a prototype of what I'm thinking (tomorrow), if that works?

Reply:
> I'm happy to test out a prototype of what I'm thinking

Spent 15 minutes and I do think this'll be simpler. Dealing with symmetric padding around focused tips is still a TODO in my version however. https://github.com/nextstrain/auspice/commit/ec13e23b3657f499fee2b254f2854b27a4b9f6b3

Reply:
And then spent another bit of time getting the symmetry working... I'll stop now.

6074280535c6c9a104a7d1b5d3ef850641fe3716 (on top of https://github.com/nextstrain/auspice/commit/ec13e23b3657f499fee2b254f2854b27a4b9f6b3 above) accomplishes it using a closure to capture `previousWasVisible`, just as you did.

https://github.com/nextstrain/auspice/commit/80fe5386d8c5a445117e2f9310514ee2d538fdae uses a stack to traverse the tree rather than a recursive function. It's probably better and faster in the long run, but probably has bugs and so I wouldn't use it in this PR.

Reply:
Thanks!